src/redis_disk.cc

```diff
@@ -0,0 +1,146 @@
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size);
+}
+
+rocksdb::Status Disk::GetListSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    ListMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisList, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    std::string buf;
+    PutFixed64(&buf, metadata.head);
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size, buf);
+}
+
+rocksdb::Status Disk::GetZsetSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    ZSetMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisZSet, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    std::string score_bytes;
+    PutDouble(&score_bytes, kMinScore);
+    s = this->GetApproximateSizes(metadata, ns_key,
+                                  storage_->GetCFHandle(Engine::kZSetScoreColumnFamilyName),
+                                  key_size, score_bytes, score_bytes);
+    if (!s.ok()) return s;
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size);
+}
+
+rocksdb::Status Disk::GetBitmapSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    BitmapMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisBitmap, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    std::string score_bytes;
+    PutDouble(&score_bytes, kMinScore);
+    return this->GetApproximateSizes(metadata, ns_key,
+                                     storage_->GetCFHandle(Engine::kSubkeyColumnFamilyName),
+                                     key_size, std::to_string(0), std::to_string(0));
+}
+
+rocksdb::Status Disk::GetSortedintSize(const Slice &user_key, uint64_t *key_size) {
+    *key_size = 0;
+    std::string ns_key;
+    AppendNamespacePrefix(user_key, &ns_key);
+    SortedintMetadata metadata(false);
+    rocksdb::Status s = Database::GetMetadata(kRedisSortedint, ns_key, &metadata);
+    if (!s.ok()) return s.IsNotFound() ? rocksdb::Status::OK() : s;
+    std::string prefix_key, next_version_prefix_key, start_buf;
+    PutFixed64(&start_buf, 0);
+    return this->GetApproximateSizes(metadata, ns_key,
```

Comment:
You can omit `this` when calling class functions here and in other functions as well.

Reply:
> You can omit `this` when calling class functions here and in other functions as well.

thank you. It has been modified.